pytab/core.py

```diff
@@ -146,44 +146,55 @@ def unpackArchive(archive_path, target_path):
     click.echo(" success!")
 
 
-def benchmark(ffmpeg_cmd: str) -> tuple:
+def benchmark(ffmpeg_cmd: str, debug_flag: bool, prog_bar) -> tuple:
     runs = []
     total_workers = 1
     run = True
-    last_Speed = -0.5  # to Assure first worker always has the required difference
+    last_speed = -0.5  # to Assure first worker always has the required difference
+    formatted_last_speed = "00.00"
     failure_reason = []
-
-    with click.progressbar(
-        length=0, label=f"Workers: {total_workers}, Speed: 0.0"
-    ) as progress_bar:
-        while run:
-            output = worker.workMan(total_workers, ffmpeg_cmd)
-            # First check if we continue Running:
-            if output[0]:
-                run = False
-                failure_reason.append(output[1])
-            elif output[1]["speed"] < 1:
-                run = False
-                failure_reason.append("performance")
-            # elif output[1]["speed"]-last_Speed < 0.5:
-            #    run = False
-            #    failure_reason.append("failed_inconclusive")
-            else:  # When no failure happened
-                runs.append(output[1])
-                total_workers += 1
-                last_Speed = output[1]["speed"]
-                progress_bar.label = f"Workers: {total_workers}, Speed: {last_Speed}"
-            progress_bar.update(1)
-        progress_bar.update(1)
+    if debug_flag:
+        click.echo(f"> > > > Workers: {total_workers}, Last Speed: {last_speed}")
+    while run:
+        if not debug_flag:
+            prog_bar.label = f"Testing | Workers: {total_workers:02d} | Last Speed: {formatted_last_speed}"
+            prog_bar.render_progress()
+        output = worker.workMan(total_workers, ffmpeg_cmd)
+        # First check if we continue Running:
+        if output[0]:
+            run = False
+            failure_reason.append(output[1])
+        elif output[1]["speed"] < 1:
+            run = False
+            failure_reason.append("performance")
+        # elif output[1]["speed"]-last_speed < 0.5:
+        #    run = False
+        #    failure_reason.append("failed_inconclusive")
```

Comment:
Please remove unused code

Reply:
In fact the linter agrees with you.

This code is only commented out atm, because it used to create issues when worker.py wasnt implemented.
It SHOULD work again now. But no testing has gone into it.

It will be removed or reactivated in the next few™️ days/weeks though.

I would not like to include this into the Progress Bar topic, as this is mostly UI/UX, while this code is actualy part of the Core script.

Kind regards.